pelican3.7.0

jerryasher opened the pull request:
I believe this will fix #38 reported by @szhorvat.

I'm a bit of a noob to python, git, github, pelican, mathjax, but it seems to work and the unittest seem to work:

```bash
$ python test_math.py
Traceback (most recent call last):
  File "test_math.py", line 3, in <module>
    from .render_math import parse_tex_macros, _parse_macro, _filter_duplicates
ValueError: Attempted relative import in non-package
(py2) Dell15:~/d/workspace/upstream.pelican/pelican_plugin-render_math (pelican3.7.0)
$ python test_math.py
...WARNING: macros where defined more than once, the last definition is used
Macro circ defined in
/home/user/example1.tex, line 1
/home/user/example2.tex, line 1

.WARNING: macros where defined more than once, the last definition is used
Macro circ defined in
/home/user/example.tex, line 1
/home/user/example.tex, line 2

.
----------------------------------------------------------------------
Ran 5 tests in 0.000s

OK
```

To get the unit tests to run in Python 2.7, I did have to make one change, which is not included here, to the import statement in render_math.py, removing the "." relative import.

this is, I temporarily changed
```python
try:
    from . pelican_mathjax_markdown_extension import PelicanMathJaxExtension
except ImportError as e:
    PelicanMathJaxExtension = None
```

to
```python
try:
    from pelican_mathjax_markdown_extension import PelicanMathJaxExtension
except ImportError as e:
    PelicanMathJaxExtension = None
```

Commit:
pel 3.7 change MD_EXTENSIONS to MARKDOWN and _get_summary to summary

Commit:
pel 3.7 change MD_EXTENSIONS to MARKDOWN and _get_summary to summary

szhorvat commented:
Yes, it does appear to fix the problem.

There's still the (separate) _get_summary deprecation in Pelican 3.7, but that's unrelated.

jerryasher commented:
@szhorvat I think this new version of the patch/pull request should remove the deprecation issue, which I fixed, tested, added, committed, then squashed the two commits together.

I am absolutely certain you use this plugin more than I, so if you could test it and make sure it works that would be great.

szhorvat commented:
I tried it again.  It appears to work fine both with Pelican 3.7.1 and 3.6.3.  Thank you for the fix!

szhorvat commented:
Actually, I just noticed a problem. With the 

    RELATIVE_URLS=False

setting, the usual script that enables MathJax doesn't get inserted into the output. This was not a problem with 3.6.3.

szhorvat commented:
To be more precise, it doesn't get inserted only on the summary pages.  The blog post pages are fine.  This is not due to the change you made, it doesn't work without that change either.  I guess I should open a new bug for this.

szhorvat commented:
I opened #40 for this other problem.

Commit:
Merge branch 'pelican3.7.0' of github.com:jerryasher/pelican_plugin-render_math into pelican3.7.0

DEKHTIARJonathan commented:
Thanks a lot for the PR!
